### PR TITLE
feat: upgrade exposed API from BigInt to BI

### PR DESCRIPTION
**Pull request by sighWang**
We have previously released a library [@ckb-lumos/bi](https://github.com/nervosnetwork/lumos/pull/226) for handling big numbers by wrapping JSBI to avoid the compatibility issues caused by `bigint`. 

This PR uses `@ckb-lumos/bi` to upgrade the lumos API that was previously exposed to the public using `bigint` in a compatible way, see the [migration suggestion](https://github.com/nervosnetwork/lumos/blob/285539de161f4873f424943c7d33fb1bb15310f8/documentations/migration-from-bigint-to-bi.md) for details of how to do this and how to migrate.

### CODE REVIEW

File: packages/base/lib/index.js [hunk 1]
```diff
@@ -7,4 +9,6 @@ module.exports = {
   helpers: require("./helpers"),
   indexer: require("./indexer"),
   logger: require("./logger"),
+  primitive,
+  ...primitive,
```
Comment (homura) [inline packages/base/lib/index.js]: we should not expose primitive since JSBI is NOT a primitive anymore

File: packages/base/lib/primitive.d.ts [hunk 2]
```diff
@@ -1,17 +1,23 @@
+import primitive from "jsbi";
```
Comment (homura) [inline packages/base/lib/primitive.d.ts]: can we remove `jsbi` in primitive?

File: packages/base/lib/primitive.js [hunk 3]
```diff
@@ -1 +1,84 @@
 // for typescript
+const JSBI = require("jsbi");
```
Comment (homura) [inline packages/base/lib/primitive.js]: can we remove JSBI ?

File: packages/base/lib/since.js [hunk 4]
```diff
@@ -89,98 +99,110 @@ function generateAbsoluteEpochSince({ length, index, number }) {
...
+      if (
+        !BI.from(added.length).eq(0) &&
+        BI.from(added.index).gte(added.length)
+      ) {
+        let _x, _y;
+
+        (_x = added),
+          (_y = "number"),
+          (_x[_y] = BI.from(added.index).div(added.length).add(_x[_y]));
+        added.index = BI.from(added.index).mod(added.length);
```
Comment (homura) [inline packages/base/lib/since.js]: would you please refactor the `_x` `_y` for more readable

**Comment by homura** [comment]
would you please make a PR to the `ckb2021` branch?

File: packages/base/lib/primitive.js (hunk 3 above)
Comment (sighWang) [reply packages/base/lib/primitive.js]: can we remove primitve.js?
cause there is no code use primitive.js and BI package can do everything primitive.js can.

File: packages/base/lib/index.js (hunk 1 above)
Comment (sighWang) [reply packages/base/lib/index.js]: deleted

File: packages/base/lib/primitive.d.ts (hunk 2 above)
Comment (sighWang) [reply packages/base/lib/primitive.d.ts]: deleted

File: packages/base/lib/primitive.js (hunk 3 above)
Comment (sighWang) [reply packages/base/lib/primitive.js]: remove primitive.js code

File: packages/base/lib/since.js (hunk 4 above)
Comment (sighWang) [reply packages/base/lib/since.js]: https://github.com/nervosnetwork/lumos/pull/248/commits/9f1b8fdbe6fcc08eef2873a1f1386940736cbeba

**Pull request merged**